Blog post: special interest groups (#2343)
nf-core/website: src/content/blog/2024/special_interest_groups.mdx

---
title: Special Interest Groups
subtitle: A new place to collaborate with others working in your field
pubDate: 2024-03-02T00:00:00+01:00
#pubDate: 2024-03-12T00:00:00+01:00
headerImage: https://images.unsplash.com/photo-1483428400520-675ef69a3bc4
headerImageAlt: 'Photo by Hudson Hintze on Unsplash'
authors:
    - 'ewels'
label:
    - 'community'
embedHeaderImage: false
---

> import cross_functional from '@assets/images/blog/special_interest_groups/cross-functional-organisation.png';

Reviewer: ```suggestion
import crossFunctional from '@assets/images/blog/special_interest_groups/cross-functional-organisation.png';
```
🐫

import { Image } from 'astro:assets';

Today, we're very happy to announce a new nf-core initiative:
_Special Interest Groups_.

# Background

> Until now, the nf-core has been very focussed on doing what we do best:

Reviewer: ```suggestion
Until now, nf-core has been very focused on doing what we do best:
```

> creating analysis pipelines and the tooling around them.

Reviewer: this whole paragraph reads bumpy

Author: Yeah, I ended up deleting most of it.

> We have structured ourselves around this: Slack channels for every pipeline
> bring together people working on the same set of tools to collaborate.
> It's clear to newcomers where to find help and it keeps the discussion focussed.

Reviewer 2: I tried to rephrase it a little from @mashehu comment 
> this whole paragraph reads bumpy
```suggestion
Until now, we have structured the nf-core community around this: Slack channels for every pipeline that
bring together people working on the same set of tools to collaborate.
This, makes clear to newcomers where to find help and also keeps the discussion focussed.
```

Author: Thanks! I ended up doing a major carve and deleted most of it in the end.

However, as the community grows, the tool-specific organisation is beginning to show a few gaps.

> Different research projects can use the _same_ set of pipelines in very different ways:

Reviewer 2: ```suggestion
Researchers from different disciplines can use the _same_ set of pipelines in very different ways:
```

for example [nf-core/sarek](/sarek) was primarily written for analysis of somatic

> cancer sequencing analysis, with tumor:normal Human DNA samples as inputs.

Reviewer 2: ```suggestion
cancer sequencing analysis, with `tumor:normal` Human DNA samples as inputs.
```
or otherwise with quotes:
```
with "tumor:normal" Human DNA samples as inputs.
```

> However, it works great as a generic variant calling pipeline and indeed many groups use

Reviewer 2: ```suggestion
However, Sarek also works great as a generic variant calling pipeline and indeed many groups use
```

it with DNA sequencing data from a diverse range of organisms, even bacteria!
Most nf-core pipelines are very flexible in their usage by design, and so for truly
reproducible research groups need to standardise not just the pipelines that they
use, but also the manner in which they use them.

> We see more and more large research consortia adopting nf-core pipelines for their

Reviewer 2: ```suggestion
Additionally, many of these researchers actively contribute to major research consortia. As a result, we see more and more large research consortia adopting nf-core pipelines for their
```

research and these concerns are often key to their desire to make research outputs
from disparate groups comparable and reproducible.

To meet this need, we want to provide a space for community members working on
similar research interests to gather and collaborate.
We want to foster collaboration not just in the development of scientific
research tooling, but also in the usage of that tooling across multiple fields.
We hope that special interest groups will provide a place for this to happen.

# How will this work?

We like how our community is structured around pipelines.
Our established methods for these groups appears to work well, so rather than
reinventing the wheel we intend to adopt a similar pattern for special interest groups:

1. Anyone can suggest a new group in the `#new-interest-groups` Slack channel
    - Just like with `#new-pipelines`, there is a Slack workflow with a form to fill in,
      with a handful of questions to collect all the required information.
2. The core team (and steering group, as necessary) will discuss the proposal, suggest
   any changes and either accept or reject the proposal.
    - This process will be tracked in a GitHub project, called

>       [New interest groups](https://github.com/orgs/nf-core/projects/63)

Reviewer 2: ```suggestion
      [New interest groups](https://github.com/orgs/nf-core/projects/63).
```

> 3. A new Slack channel will be created for the new interest group on the nf-core Slack
> 4. A new web page will be created for the group on the nf-core website

Reviewer 2: ```suggestion
3. A new Slack channel will be created for the new interest group on the nf-core Slack.
4. A new web page will be created for the group on the nf-core website.
```

5. The lead for the new group will organise a nf-core/bytesize talk to introduce
   the group to the community. A blog post will also be recommended.
6. The new special interest group will be active and continue under their own steam.

## Group web pages

> The new web page will just be a stub at first, but we hope that the interest group

Reviewer 2: Is this ready? If yes would be nice to include a pointer to where this will live.

sub-pages, including things like best-practice documentation, meeting notes and more.

We would like to standardise a listing of the nf-core pipelines typically used by
members of each interest group, so that the pages can be cross-linked.
That way, anyone viewing a web page for an nf-core pipeline will be able to see
at a glance what it can be used for and hopefully link to best practices for
that pipeline with that type of data.

## Closing after inactivity

Just like pipelines, special interest groups will never be _deleted_.
As part of the annual "spring cleaning" that the nf-core maintainers team does,
we will check on each group's recent activity and make sure that everything is going ok.
If a group is obviously inactive and does not respond to requests for a status update,
or if desired by the organisers, the interest group will be archived.

Anyone will be able to revive an archived special interest group at a future date,
in a process much like proposing a new group.

## Who can join?

Anyone can join any interest group.
Just as every nf-core pipeline is owned by the community, special interest groups

> will be made as inclusive as possible.

Reviewer 2: I will join the two paragraphs as the one below is a consequence of the one above with something such as:
"""
Hence, we will try to avoid...
"""

We will try to avoid naming groups after existing groups or consortia, though
many may begin from such origins. We will also try to not to tie any interest
groups to specific topics or research groups where possible.
We want anyone working in a relevant field or interest to be able to join (and leave!)
an interest group without feeling like an outsider.

## What will they do?

We fully expect every interest group to be different: both in their scope and
membership, but also in the time that their organisers can commit and the activities
that would be most useful to their members.
As such, we want to be as flexible as possible with our expectactions.
If the group is active and people are finding it useful, we will consider it a success.

> That said, the types of things that we hope that interest groups could do could include:

Reviewer 2: Should we include something such as "contribute to pipelines developing the specific tooling for their research field if needed"?

> -   Writing best practices for nf-core pipeline usage, specific to their field
> -   Creating pipeline configs specific to the group, to encode these recommendations
> -   Work together to harmonise usage of data formats
> -   Listing relevant externala and internal resources, videos and documentation
> -   Holding regular group meetings
> -   Presenting updates in nf-core/bytesize talks and blog posts
> -   Potentially even running their own series of talks, if there is interest

Reviewer 2: ```suggestion
-   Writing best practices for nf-core pipeline usage, specific to their field.
-   Creating pipeline configs specific to the group, to encode these recommendations.
-   Work together to harmonise usage of data formats.
-   Listing relevant externala and internal resources, videos and documentation.
-   Holding regular group meetings.
-   Presenting updates in nf-core/bytesize talks and blog posts.
-   Potentially even running their own series of talks, if there is interest.
```

We're sure that our new groups can and will do many more different and diverse things.
We can't wait to see where the program goes!

## What about pipeline development?

Discussion within special interest groups can revolve around usage of pipelines,
but when it comes to any missing features or requirements we expect members to move
discussions into the regular pipeline channels.
There should be a distinction between scientific and technical topics.

In this way, we hope that interest groups and pipeline groups work in an
orthogonal manner, each contributing to the other.

> <Image src={cross_functional} alt="Cross-functional pipelines and interest group structure" />

Reviewer: ```suggestion
<Image src={crossFunctional} alt="Cross-functional pipelines and interest group structure" />
```

# Introducing: Our first special interest group

> Today we are delighted to announce our first interest group: _Domestic animal genomics_.

Reviewer 2: ```suggestion
Today we are delighted to announce our first interest group: _Animal genomics_.
```

We'll leave the details to be elaborated on in a dedicated blog post by them at a later
date, but we'd like to take this opportunity to thank them for taking the lead.

This interest group has formed out of the [BovReg](https://bovreg.eu/) and
[EuroFAANG](https://eurofaang.eu/) research consortia.
BovReg has been working closely with nf-core since its inception and has helped
bring other EuroFAANG consortia members together to use nf-core pipelines,

> including GEroNIMO, AQUA-FAANG, GENE-SWitCH, Rumigen and HoloRuminant.

Reviewer 2: I added the links to the projects and ordered them alphabetical to avoid any 😡 
```suggestion
including [AQUA-FAANG](https://www.aqua-faang.eu/), [EuroFAANG Research Infrastructure](https://eurofaang.eu/), [GENE-SWitCH](https://www.gene-switch.eu/), [GEroNIMO](https://www.geronimo-h2020.eu/), [HoloRuminant](https://holoruminant.eu/) and [Rumigen](https://rumigen.eu/).
```
